### DISCUSSION TITLE
We're having issues with the Composit and Content Search API for newly created Primary Category

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Knowlg/Community#201] [General Discussions]
Hello @AmiableAnil @pallakartheekreddy,

**Issue:** 

We're having issues with the Composit and Content Search API. Newly created data under the event primary category cannot be discovered.

**Details:**

I set up a new objectCategoryDefination (event_event_all) and master category (event). Added new data to it.

After using the read API for the event, I can see the entire data like this:

```
"identifier": "do_114035725939867648133",
"name": "Event 7th",
"objectType": "Event",
"primaryCategory": "Event",
"contentType": "Event",
"visibility": "Default",
"status": "Live"
```

and so on.

However, neither the content nor the composition of the search APIs contain this data.

**Content Search API:** 

`{{host}}/api/content/v1/search
`

**Body:** 

```
{
    "request": {
        "filters": {
            "primaryCategory": [
                "Event"
            ]
        }
    }
}
```

**Composite Search API:** 

`{{host}}/action/composite/v3/search
`

**Body:** 

```
{
    "request": {
        "query": "",
        "filters": {
            "primaryCategory": [],
            "status": [
                "Live"
            ]
        },
        "sort_by": {
            "lastUpdatedOn": "desc"
        }
    }
}
```

Please assist us in this matter.

cc @vaivk369, @baljeetsingh, @manishakapadnis

**Comment by komalm** [comment]
Hello @pallakartheekreddy,

I hope this message finds you well. I wanted to follow up on the issue we reported regarding the Composite and Content Search APIs not discovering newly created data under the "event" primary category.

As per our previous communication on this [ticket](https://github.com/orgs/Sunbird-Knowlg/discussions/198), we provided detailed information about the problem, including the setup of a new objectCategoryDefinition and master category, along with examples of the data not being retrieved by the search APIs.

Could you please provide an update on the progress made in investigating and resolving this issue? Additionally, if there are any further details needed from our end or any actions we can take to assist with troubleshooting, please let us know.

Thank you for your attention to this matter, and I look forward to hearing from you soon.

cc @vaivk369, @baljeetsingh, @manishakapadnis

**Reply by pallakartheekreddy** [reply]
@komalm Could you kindly provide further details on the following:
1. Could you confirm whether the search API service and search-indexer Flink jobs are currently operational?
2. Have any errors been detected in the search-indexer Flink jobs?

**Reply by komalm** [reply]
@pallakartheekreddy We have restarted it and this is working fine. we are not able to find any blocker there.

**Comment by pallakartheekreddy** [comment]
@komalm Closing this thread as the issue has been resolved. If you have any further questions or concerns, please feel free to reopen the thread.

**Reply by komalm** [reply]
@pallakartheekreddy Sorry, The issue is still open. 

I have given a reply to your question.:

We have already restarted it and this is working fine. we are not able to find any blocker there. 

But the issue is still open.

**Reply by pallakartheekreddy** [reply]
@komalm Can we get into a call to check this issue?

**Reply by komalm** [reply]
@pallakartheekreddy Can we schedule a meeting for 3:30 today?

**Reply by pallakartheekreddy** [reply]
@komalm Yeah, works for me.

**Reply by komalm** [reply]
Hello @pallakartheekreddy,

Thank you for providing an update on the issue. It's great to see that the problem has been resolved by removing the `Event` and `EventImage` values under the `restrict.objectTypes` from the `search-indexer-config`.

With this adjustment, it's excellent to know that the event is now discoverable in the composite search API.

Thank you to @pallakartheekreddy for your valuable guidance and support throughout this process.

**Reply by pallakartheekreddy** [reply]
@komalm Closing this thread as the issue has been resolved. If you have any further questions or concerns, please feel free to reopen the thread.
